File: PlacementCellBackend/Services/Placements/Interfaces/IPlacementDriveService.cs
using PlacementCellBackend.DTOs.PlacementDriveDTOs;

namespace PlacementCellBackend.Services.Placements.Interfaces
{
    public interface IPlacementDriveService
    {
        Task<IEnumerable<PlacementDriveDTO>> GetAllPlacementDrivesAsync();
        Task<IEnumerable<PlacementDriveDTO>> GetPlacementDrivesByCollegeAsync(int collegeId);
        Task<IEnumerable<PlacementDriveDTO>> GetPlacementDrivesByCompanyAsync(string companyId);
        Task<IEnumerable<PlacementDriveDTO>> GetPlacementDrivesByDateAsync(DateOnly date);
        Task<IEnumerable<PlacementDriveDTO>> GetUpcomingPlacementDrivesAsync();
        Task<PlacementDriveDTO?> GetPlacementDriveByIdAsync(int driveId);
        Task<PlacementDriveDTO> SchedulePlacementDriveAsync(CreatePlacementDriveDTO driveDto);
        Task<bool> UpdatePlacementDriveAsync(int driveId, UpdatePlacementDriveDTO driveDto);
        Task<bool> UpdateDriveStatusAsync(int driveId, string status);
        Task<bool> CancelPlacementDriveAsync(int driveId);
        Task<bool> DeletePlacementDriveAsync(int driveId);
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/Placements/Interfaces/IPlacementDriveService.java
package com.example.placementicsbackend.services.placements.interfaces;

import com.example.placementicsbackend.dtos.PlacementDriveDTOs.CreatePlacementDriveDto;
import com.example.placementicsbackend.dtos.PlacementDriveDTOs.PlacementDriveDto;
import com.example.placementicsbackend.dtos.PlacementDriveDTOs.UpdatePlacementDriveDto;
import java.time.LocalDate;
import java.util.List;

public interface IPlacementDriveService {

    List<PlacementDriveDto> getAllPlacementDrives();

    List<PlacementDriveDto> getPlacementDrivesByCollege(int collegeId);

    List<PlacementDriveDto> getPlacementDrivesByCompany(String companyId);

    List<PlacementDriveDto> getPlacementDrivesByDate(LocalDate date);

    List<PlacementDriveDto> getUpcomingPlacementDrives();

    PlacementDriveDto getPlacementDriveById(int driveId);

    PlacementDriveDto schedulePlacementDrive(CreatePlacementDriveDto driveDto);

    boolean updatePlacementDrive(int driveId, UpdatePlacementDriveDto driveDto);

    boolean updateDriveStatus(int driveId, String status);

    boolean cancelPlacementDrive(int driveId);

    boolean deletePlacementDrive(int driveId);
}
